crates/ruff/src/args.rs

```diff
@@ -806,15 +806,18 @@ impl TypedValueParser for ConfigArgumentParser {
         arg: Option<&clap::Arg>,
         value: &std::ffi::OsStr,
     ) -> Result<Self::Value, clap::Error> {
-        let path_to_config_file = PathBuf::from(value);
-        if path_to_config_file.exists() {
-            return Ok(SingleConfigArgument::FilePath(path_to_config_file));
-        }
-
         let value = value
             .to_str()
             .ok_or_else(|| clap::Error::new(clap::error::ErrorKind::InvalidUtf8))?;
 
+        if let Ok(path_to_config_file) =
+            shellexpand::full(value).map(|config| PathBuf::from(config.as_ref()))
```

Comment:
```suggestion
            shellexpand::full(value).map(|config| PathBuf::from(&config))
```

Reply:
You'll be surprised to learn that this doesn't work, because `config` is `Cow<Xstr>`.

Reply:
You just need to find the right number of starts: Maybe `&**config`?